test/core/test_metric_comparison.py

```diff
@@ -134,3 +157,79 @@ def test_multiple_values_same_metric(self):
         self.assertAlmostEqual(true_mean, mean)
         self.assertAlmostEqual(true_stddev, stddev)
         self.assertEqual(2, count)
+
+    def test_basic(self):
+
+        #   metric full name            | build1 result | build2 result | expected result
+        # False -> the metric has regressed, True -> the metric has been fixed, None -> not a regression nor fix
+        test_cases = {
+            'suite_a/improved-metric-higher-better': (1, 2, True),
+            'suite_a/improved-metric-lower-better': (2, 1, True),
+            'suite_a/regressing-metric-higher-better': (2, 1, False),
+            'suite_a/regressing-metric-lower-better': (1, 2, False),
+            'suite_a/stable-metric': (1, 1, None),
+            'suite_a/thresholdless-metric': (1, 2, None),
+            'suite_a/valueness-threshold-metric': (1, 2, None),
+        }
+
+        for metric_name in test_cases.keys():
+            build_a_result = test_cases[metric_name][0]
```

Comment:
I think this could be unpacked like this:

```
build_a_result, build_b_result, expected = test_cases[metric_name]
```

But the current code may be more verbose.